### PR TITLE
feat(topology): add affinity key to core

### PR DESCRIPTION
**Pull request by sinhaashish**

### CODE REVIEW

File: control-plane/agents/src/bin/core/controller/scheduling/volume.rs [hunk 1]
```diff
@@ -118,6 +122,103 @@ impl Deref for GetSuitablePools {
     }
 }
 
+/// Create replicas of a volume.
+/// Selects the best pool candidates to create lvol replicas on.
```
Comment (tiagolobocastro) [inline control-plane/agents/src/bin/core/controller/scheduling/volume.rs]:
```suggestion
/// Selects the best pool candidates to create volume replicas on.
```

File: control-plane/agents/src/bin/core/controller/scheduling/volume.rs [hunk 2]
```diff
@@ -118,6 +122,103 @@ impl Deref for GetSuitablePools {
...
+pub(crate) struct CreateVolumeReplica {
```
Comment (tiagolobocastro) [inline control-plane/agents/src/bin/core/controller/scheduling/volume.rs]: Why do we need a new type?  Moreover it's quite similar to `AddVolumeReplicas`

File: control-plane/grpc/proto/v1/pool/pool.proto [hunk 3]
```diff
@@ -64,6 +64,8 @@ message PoolState {
   uint64 used = 6;
   // total size of pool replicas
   optional uint64 committed = 7;
+  // optional labels of pool
```
Comment (tiagolobocastro) [inline control-plane/grpc/proto/v1/pool/pool.proto]: hmm this doesn't make sense? Dataplane pool state has no notion of labels...

File: control-plane/agents/src/bin/core/volume/scheduling.rs [hunk 4]
```diff
@@ -15,9 +15,10 @@ use crate::{
     node::wrapper::NodeWrapper,
 };
 use agents::errors::{NotEnough, SvcError};
+use std::collections::HashMap;
```
Comment (Abhinandan-Purkait) [inline control-plane/agents/src/bin/core/volume/scheduling.rs:18]: Please move this down after a blank line.

**Comment by sinhaashish** [comment]
PR https://github.com/openebs/mayastor-control-plane/pull/846 raised for the same.

**Pull request closed without merging**